src/libcore/iter/iterator.rs

```diff
@@ -258,6 +258,40 @@ pub trait Iterator {
         None
     }
 
+    /// Creates an iterator starting at the same point, but stepping by
+    /// the given amount at each iteration.
+    ///
+    /// Note that it will always return the first element of the range,
+    /// regardless of the step given.
+    ///
+    /// # Panics
+    ///
+    /// If the given step is `0`, the method will panic if debug assertions are
+    /// enabled.
```

Comment:
nit: This comment makes me think `debug_assert!`, but it's not.  Does it need more than "This method will panic if the given `step` is `0`."?